adding warnings for scenarios 5 and 6 (#5591)
mlflow/mlflow: docs/source/tracking.rst

  * Retrieving artifacts from the configured backend store for a user request is done with the same authorized authentication that was configured at server start
  * Artifacts are passed to the end user through the Tracking Server through the interface of the ``HttpArtifactRepository``

.. note::
    When an experiment is created, the artifact storage location from the configuration of the tracking server is logged in the experiment's metadata.
    If enabling proxied artifact storage, any existing experiments that were created while operating a tracking server in non-proxy mode will not

>     be available for adding runs to them. This is due to the default artifact storage locations (``artifact_uri``) of the old experiments logged prior to

Reviewer: ```
If enabling proxied artifact storage, any existing experiments that were created while operating a tracking server in non-proxy mode will not
    be available for adding runs to them
```

This isn't quite accurate, right? Clients can still log to the old experiment as long as they have sufficient credentials.

Author: That definitely could have been worded more clearly and listing out details instead of assuming why someone would want to use the tracking server for proxy serving. I reworded this to focus on this security aspect and why creating new experiments might be the best after migrating to this config.

>     If enabling proxied artifact storage, any existing experiments that were created while operating a tracking server in non-proxy mode will not
>     be available for adding runs to them. This is due to the default artifact storage locations (``artifact_uri``) of the old experiments logged prior to
>     enabling a tracking server's proxied access while operating in  ``--serve-artifacts`` mode. It is highly advised to create new experiments when performing
>     a migration that enables proxied artifact access to prevent errors in logging run artifacts to an existing experiment from occurring at the client side.

Reviewer: ```suggestion
    When enabling proxied artifact storage, any existing experiments that were created while operating a tracking server in non-proxied mode will continue to use a non-proxied artifact location. In order to use proxied artifact logging, a new experiment must be created.
```

Author: Inserted in-line

  * Listing of artifact responses will pass from the file store through the Tracking Server to the client
  * Loading of artifacts will utilize the access credentials of the MLflow Tracking Server to acquire the files which are then passed on to the client

.. note::
  - If migrating from Scenario 5 to Scenario 6 due to request volumes, it is important to perform two validations:

>     - Ensure that the new tracking server that is operating in ``--artifacts-only`` mode has access permissions to the

Reviewer: This looks good! On the subject of `--artifacts-only`, I now feel that https://mlflow.org/docs/latest/tracking.html tightly couples the description of proxied logging and `--artifacts-only`, despite this being a less common use case. Could we update the example in https://mlflow.org/docs/latest/tracking.html#mlflow-tracking-servers to just use `--serve-artifacts` without `--artifacts-only` and also update https://mlflow.org/docs/latest/tracking.html#using-the-tracking-server-exclusively-for-proxied-artifact-access so that the section talks about artifact proxying with `--serve-artifacts` (without `--artifacts-only`) and includes a *subsection* discussing `--artifacts-only`? I realize that broadens the scope of this PR, but it should make the content more accessible for users.

Author: Great ideas! I de-emphasized the --artifacts-only topics, separated it out into a sub-discussion (smaller header) and focused on the elements that users have expressed confusion on. Examples updated! TY for the great feedback on this - it reads MUCH better now.
